libexec/alluxio-config.sh

```diff
@@ -32,13 +32,34 @@ ALLUXIO_HOME=$(dirname $(dirname "${this}"))
 ALLUXIO_ASSEMBLY_CLIENT_JAR="${ALLUXIO_HOME}/assembly/client/target/alluxio-assembly-client-${VERSION}-jar-with-dependencies.jar"
 ALLUXIO_ASSEMBLY_SERVER_JAR="${ALLUXIO_HOME}/assembly/server/target/alluxio-assembly-server-${VERSION}-jar-with-dependencies.jar"
 ALLUXIO_CONF_DIR="${ALLUXIO_CONF_DIR:-${ALLUXIO_HOME}/conf}"
-ALLUXIO_LOGS_DIR="${ALLUXIO_LOGS_DIR:-${ALLUXIO_HOME}/logs}"
-ALLUXIO_USER_LOGS_DIR="${ALLUXIO_USER_LOGS_DIR:-${ALLUXIO_LOGS_DIR}/user}"
 
 if [[ -e "${ALLUXIO_CONF_DIR}/alluxio-env.sh" ]]; then
   . "${ALLUXIO_CONF_DIR}/alluxio-env.sh"
 fi
 
+# This will set the value of ALLUXIO_LOGS_DIR based in the user configuration.
+# the environment variable ${ALLUXIO_LOGS_DIR} will be firstly consider.
+ALLUXIO_LOGS_DIR_JAVA_OPT_PRE="-Dalluxio.logs.dir="
+if [[  -n "${ALLUXIO_LOGS_DIR}" ]]; then
+  # If user set ALLUXIO_LOGS_DIR, use this.
+  ALLUXIO_LOGS_DIR="${ALLUXIO_LOGS_DIR}"
+elif [[ ${ALLUXIO_JAVA_OPTS} == *${ALLUXIO_LOGS_DIR_JAVA_OPT_PRE}* ]]; then
+  # If user set alluxio.logs.dir in ALLUXIO_JAVA_OPTS, use this.
+  # Extract "XXX -Dalluxio.logs.dir=<value> XXX".
+  # First get "<value> XXX".
+  ALLUXIO_LOGS_DIR_TMP=${ALLUXIO_JAVA_OPTS#*${ALLUXIO_LOGS_DIR_JAVA_OPT_PRE}}
+  # Then get "<value>".
+  ALLUXIO_LOGS_DIR=$(echo "${ALLUXIO_LOGS_DIR_TMP}"|awk '{print $1}')
+  # Set ALLUXIO_LOGS_DIR_FROM_ALLUXIO_JAVA_OPTS_TAG to true is let ALLUXIO_JAVA_OPTS
+  # not append -Dalluxio.user.logs.dir=<value> later, since it already have.
+  ALLUXIO_LOGS_DIR_FROM_ALLUXIO_JAVA_OPTS_TAG="true"
```

Comment:
I haven't give it too much thought, but I believe the logic doesn't have to be this complicated?

The issue is the same for all environment var translations. For example if you define `ALLUXIO_WORKER_HOSTNAME=a` but `-Dalluxio.worker.hostname=b`, we have the line `ALLUXIO_WORKER_JAVA_OPTS+=-Dalluxio.worker.hostname=${ALLUXIO_WORKER_HOSTNAME}` so one overrides the other. For alluxio, it often goes without saying that you use the env var and rely on them being translated to a java property somewhere.

I think one easy way out is we ask the users to use `ALLUXIO_LOGS_DIR` instead of `-Dalluxio.logs.dir`.

Otherwise we can search if `alluxio.logs.dir` is set in java properties already so we don't set later.

Reply:
I think my modify logic is the same as 

```
I think one easy way out is we ask the users to use ALLUXIO_LOGS_DIR instead of -Dalluxio.logs.dir.

Otherwise we can search if alluxio.logs.dir is set in java properties already so we don't set later.
```
But we must set the value  of `alluxio.logs.dir` from  java property.
Because many shell scrips relay on this var such as `alluxio-start.sh`, `alluxio-master.sh` etc.
There are snippets in alluxio-start.sh
```
ensure_dirs() {
  if [[ ! -d "${ALLUXIO_LOGS_DIR}" ]]; then
    echo "ALLUXIO_LOGS_DIR: ${ALLUXIO_LOGS_DIR}"
    mkdir -p ${ALLUXIO_LOGS_DIR}
  fi
}
```
and 
```
    echo "Starting job master @ $(hostname -f). Logging to ${ALLUXIO_LOGS_DIR}"

```

So if we do not define a env var but define in java property , and I change the code to use java property actually , then the script will be all wrong.

Reply:
i do not think it is a good idea to use the env var value of ALLUXIO_LOGS_DIR in later shell scripts. the env var is not the source of truth. can we run `bin/alluxio getConf alluxio.logs.dir` to get this value when it is time to run the `mkdir -p` command?